### ISSUE TITLE
Missing function: create_context_from_uri

### ISSUE THREAD

**Issue by knightshrub** [fpagliughi/rust-industrial-io#1]
Hello, the library currently misses the `create_context_from_uri` function.
I need this in order to work with AD's ADALM-PLUTO SDR.
I don't have a lot of experience using unsafe rust or doing ffi in rust, but would this be an acceptable implementation?
```rust
/// Tries to create a context from the specified URI
pub fn from_uri(uri: &str) -> Result<Context> {
    let uri = CString::new(uri).expect("CString::new(uri) failed");
    let ctx = unsafe { 
        ffi::iio_create_context_from_uri(uri.as_ptr()) 
     };
     if ctx.is_null() { bail!(SysError(Errno::last())); }
     Ok(Context{ ctx, })
}
```

**Comment by fpagliughi** [comment]
The first line is a bit problematic:
```
let uri = CString::new(uri).expect("CString::new(uri) failed");
```

 - I want to get rid of any code that panics in the library and have it return good error results.
 - The message string, if there is one, should not be so Rust specific, but rather should indicate the problem, like "Bad string used as URI". Something like that.

Submit this as a Pull Request and I'll get it in shortly. I'll be back working on this library soon.

**Comment by fpagliughi** [comment]
Oh, and thanks. First issue for the library!

**Comment by fpagliughi** [comment]
Fixed with f7cb280f46f1812339035a21d84feca797e66aac
